runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineRunner.java

```diff
@@ -213,6 +299,118 @@ public PortablePipelineResult run(RunnerApi.Pipeline pipeline, JobInfo jobInfo)
             result);
     metricsPusher.start();
 
+    if (eventLoggingListener != null) {
+      HashMap<String, String> driverLogs = new HashMap<String, String>();
+      MetricResults metricResults = result.metrics();
+      for (MetricResult<DistributionResult> distributionResultMetricResult :
```

Comment:
What about the other metric types (counters and gauges)?

Reply:
Thank you, they were add in the last commit.